includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -259,6 +305,31 @@ public static function handle_newsletter_subscribed( $params, $data ) {
 		return $params;
 	}
 
+	/**
+	 * Handler for the campaign_interaction event.
+	 *
+	 * @param int   $params The GA4 event parameters.
+	 * @param array $data      Data associated with the Data Events api event.
+	 *
+	 * @return array $params The final version of the GA4 event params that will be sent to GA.
+	 */
+	public static function handle_campaign_interaction( $params, $data ) {
+		$transformed_data = $data;
+		// remove data added in the body filter.
+		unset( $transformed_data['ga_params'] );
+		unset( $transformed_data['ga_client_id'] );
+
+		unset( $transformed_data['interaction_data'] );
+		$transformed_data = array_merge( $transformed_data, $data['interaction_data'] );
+
+		unset( $transformed_data['campaign_blocks'] );
```

Comment:
When the `reader_registered` and `newsletter_subscribed` events are sent, they are populated with these campaign metadata but not sanitized like here for `campaign_interaction`.

Should some handling happen for such cases or just let the generic sanitizer handle it? This is how it's currently handled:

```
[NEWSPACK-DATA-EVENTS-GA4][Newspack\Data_Events\Connectors\GA4::log]: Parameter campaign_blocks has an invalid value. It was removed from the np_newsletter_subscribed event.
[NEWSPACK-DATA-EVENTS-GA4][Newspack\Data_Events\Connectors\GA4::log]: Parameter interaction_data has an invalid value. It was removed from the np_newsletter_subscribed event.
```

Reply:
weird, I didn't see this happening. I'm removing these 2 keys from the data and they should never be sent in tha GA event...

I'll double check

`campaign_blocks` is transformed into the `has_XX_block` params

`interaction_data` is merged and moved to the same level as root params...

Reply:
They are but only in the context of the `campaign_interaction` event, not the other ones that also end up getting populated with the same values.

Reply:
https://github.com/Automattic/newspack-plugin/pull/2337/commits/4fa17fa9cd685c9cd3042e36664a0c609c9e9c24